apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-title/edit.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import React, { Component } from 'react';
+import { __ } from '@wordpress/i18n';
+const { withSelect } = wp.data;
+const { apiFetch } = wp;
+
+/**
+ * Internal dependencies
+ */
+
+class SiteTitleEdit extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			title: null,
+			initialTitle: null,
+		};
+	}
+
+	componentDidMount() {
+		if ( this.state.initialTitle ) {
+			return;
+		}
+
+		apiFetch( { path: '/wp/v2/settings' } ).then( ( { title } ) => {
+			this.setState( { initialTitle: title } );
+		} );
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { isSaving, isPublished, isPublishing, isAutosaving } = this.props;
+
+		if ( this.isPure() || prevProps.isSaving === this.props.isSaving ) {
+			return;
+		}
+
+		if ( ( ( isSaving && isPublished ) || isPublishing ) && ! isAutosaving ) {
+			const { title } = this.state;
+
+			apiFetch( { path: '/wp/v2/settings', method: 'POST', data: { title } } ).then( () => {
+				this.setState( { initialTitle: title } );
+			} );
+
+			this.resetTitle();
+		}
+	}
+
+	setSiteTitle = ( { target: { value } } ) => {
+		this.setState( { title: value } );
+	};
+
+	resetTitle = () => {
+		const { title } = this.state;
+
+		this.setState( { initalTitle: title } );
+	};
+
+	isPure = () => {
+		const { title, initialTitle } = this.state;
+
+		return title && title.trim() === initialTitle.trim();
+	};
+
+	render() {
+		const { title, initialTitle } = this.state;
+
+		return (
+			<div className="full-site-editing__site_title_block">
```

Comment:
Ah so if this block was not dynamic, we'd automatically get the classname `wp-block-a8c-site-title`. or `wp-block-{namespace}-{block}-{name}`

The top line already disables the Calypso linter rule for this (which is naming convention for Calypso components)
```
/* eslint-disable wpcalypso/jsx-classname-namespace */
```